### ISSUE TITLE
Train CNN Classifier

### ISSUE THREAD

**Issue by michaelmunje** [mars-vision/marsvision#17]
We will train a multioutput CNN classifier using data from DeepMars (https://www.aaai.org/ocs/index.php/AAAI/AAAI18/paper/view/16040/16400). This will require revisiting the Training method in the Model class.

**Comment by dpalencia** [comment]
Added methods to the Model class and a new DataSet class to implement transfer learning with a Pytorch CNN, though we need to ensue that the output layer has the same number of classes as the Deep Mars dataset.

**Comment by dpalencia** [comment]
I transfer learned an AlexNet model with output classes corresponding to our dataset on a GPU in a Google colab notebook. Since the file is too large to track on GitHub, I am attaching a link to a dropbox folder containing the model (and any future models I might train).

This model was trained on 700 epochs and had ~90% accuracy on a test set. I will perform some more detailed evaluation in a notebook.

[Colab Link](https://colab.research.google.com/drive/1EzqUS_RyTgyCw4Ml4ykI0yt_0DSsQU4A?usp=sharing)
[Models Folder](https://drive.google.com/drive/folders/11k1MXzS5ZtYDqcRaxNiCvfWs8xtPr9Na?usp=sharing)
